packages/@lwc/engine-dom/src/custom-elements/create-custom-element.ts

```diff
@@ -92,12 +91,18 @@ export function getUpgradableConstructor(tagName: string) {
 export const createCustomElement = (
     tagName: string,
     upgradeCallback: LifecycleCallback,
-    useNativeLifecycle: boolean
+    useNativeLifecycle: boolean,
+    isFormAssociated: boolean | undefined
 ) => {
-    const UpgradableConstructor = getUpgradableConstructor(tagName);
+    const UpgradableConstructor = getUpgradableConstructor(tagName, isFormAssociated);
```

Comment:
I think we may want to just consider `formAssociated` to be `false` if `useNativeLifecycle` is false.

Otherwise the user will be in a weird state where `formAssociated` is set to `true` on the class, but their `form*` callbacks will not do anything.

We can solve this in #3929 though. It's not a big deal.